### PR TITLE
Correct history problem on listing page

### LINKED ISSUE #14821: Moving in browser history does not work with product pagination
Going backward/forward in product overview (e.g. search results) should behave as expected (actually go back/forward to the respective page)

**To Reproduce**
1. Go to http://brave-relation.demo-ms.production.prestashop.com/en/2-home
2. Click on 'Next' or '2'
3. Go backwards in browser (arrow or swipe)
4. We stay on page 2 :(

PrestaShop version: Not sure, I guess 1.7.6
PHP version: Doesn't matter

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | History back and forward wasnt working on listing page
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #14821 
| How to test?  | Go on listing category page, then use the pagination, and try to go backward or forward using your browser function

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17014)

### CODE REVIEW

**Review by matks — DISMISSED** [review]

File: themes/classic/_dev/js/listing.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ import 'velocity-animate';
 import ProductMinitature from './components/product-miniature';
 
 $(document).ready(() => {
+  let history = window.location.href;
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/listing.js]:
```suggestion
  const history = window.location.href;
```

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Do not add assets on the develop branch please :)

**Comment by NeOMakinG** [comment]
Fine now

**Review by PierreRambaud — DISMISSED** [review]

File: themes/classic/_dev/js/listing.js (hunk 1 above)
Comment (Progi1984) [reply themes/classic/_dev/js/listing.js]: @NeOMakinG It's me or the change is not done ?

Comment (Progi1984) [reply themes/classic/_dev/js/listing.js]: @NeOMakinG The change is not ever done !

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by SD1982** [comment]
The pull request correctly fix the issue . thanks @NeOMakinG

**Comment by matks** [comment]
Thank you @NeOMakinG

**Comment by rdy4ever** [comment]
Hi there. Bit of a problem with anchor links with this one. Try clicking an anchor link like `<a href="#section1">Go to section 1</a>` and you'll notice the page refreshes unnecesary. I also use fancybox on my shop and by default it ads a hash to the url when you open a modal. This causes a page refresh when you close the modal. I traced the problem back to this PR. Maybe add a condition to be on a listing page to use it?

**Comment by rdy4ever** [comment]
Hi. A bit of a problem with this PR: now clicking anchor links triggers a whole page refresh. Could you test it? We also have a problem using fancybox (it adds a hash to to url and triggers a page refresh when you close the fancybox).

I fixed it kind of the lazy way, maybe it could be done more gratiously but here is my "fix":
```
  const is_prod_list=$("#js-product-list");
  if (is_prod_list.length) {
    window.addEventListener('popstate', function(e) {
      var state = e.state;
      window.location.href = state && state.current_url ? state.current_url : history;
    });
  }
```

**Pull request merged**